core/templates/pages/creator-dashboard-page/creator-dashboard-page.component.spec.ts

```diff
@@ -360,27 +365,36 @@ describe('Creator dashboard controller', () => {
         expect(ctrl.myExplorationsView).toBe('a');
       });
 
-      it('should set explorations sorting options', function() {
+      it('should change order sort of exploration when creator change' +
+        ' exploration order sort to the current one', function() {
         expect(ctrl.isCurrentSortDescending).toBe(true);
         expect(ctrl.currentSortType).toBe('numOpenThreads');
         ctrl.setExplorationsSortingOptions('numOpenThreads');
         expect(ctrl.isCurrentSortDescending).toBe(false);
+      });
 
+      it('should sort exploration by new open threads when creator clicks' +
+        ' on sorting options', function() {
         ctrl.setExplorationsSortingOptions('new_open');
         expect(ctrl.currentSortType).toBe('new_open');
       });
 
-      it('should set subscription sorting options', function() {
+      it('should change order sort of subscription when creator change' +
+      ' subscription order sort to the current one', function() {
         expect(ctrl.isCurrentSubscriptionSortDescending).toBe(true);
         expect(ctrl.currentSubscribersSortType).toBe('username');
         ctrl.setSubscriptionSortingOptions('username');
         expect(ctrl.isCurrentSubscriptionSortDescending).toBe(false);
+      });
 
+      it('should sort exploration by new subscriber when creator clicks' +
+        ' on sorting options', function() {
         ctrl.setSubscriptionSortingOptions('new_subscriber');
         expect(ctrl.currentSubscribersSortType).toBe('new_subscriber');
       });
 
-      it('should sort subscription function', function() {
+      it('should get value sorted in subscription function by username or' +
```

Comment:
I don't really understand what is going on here -- this may need rewriting. See if you can work it out with @nithusha21. 

Also, one general note. Sometimes, you are trying to test a general behaviour, but you might need to use specific examples for that (e.g. changing the sort option to a different one). In that case, the test name should not refer to the specific example (in this case, the type of sort option used) since the test is meant to just pick a representative example from all the different possible cases. Using a specific name where the test is testing something general makes it harder to understand what the test is actually aiming to test. Does that make sense? Please let me know if it is not clear.

Reply:
Sure, I think I get it. I decided to separate this general spec in specific ones to make the code cleaner.